### ISSUE TITLE
v0.3_stable always times out, not responding

### ISSUE THREAD

**Issue by migajek** [xplicit/HyperFastCgi#35]
I've downloaded & built v0.3_stable branch, ran it like this ``mono-server-hyperfastcgi4 /applications=/:/home/...../CRM.WebHost/ /socket=tcp:127.0.0.1:9000``
it does start properly (according to the application log) however I cannot complete any request.
According to the nginx log, every request times out
``18405#0: *5206 upstream timed out (110: Connection timed out) while reading upstream, client: <my up>, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "< my hostname>"``

When I simply run default ``fastcgi-mono-server4`` instead (with the same configuration), it does work properly and no FastCGI request times out 

I should mention that my requests are being processed in hundreds of milliseconds, always less than a second, so they're not anywhere near the timeout limit.

Mono version
```
Mono JIT compiler version 3.12.1 (tarball Fri Mar  6 19:12:47 UTC 2015)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        LLVM:          supported, not enabled.
        GC:            sgen
```

**Comment by xplicit** [comment]
most possible reason: you've got misconfiguration in nginx and hyperfastcgi keepalive settings. See this page for more info. https://github.com/xplicit/HyperFastCgi/wiki/Nginx-configuration

**Comment by xplicit** [comment]
And this related issue: https://github.com/xplicit/HyperFastCgi/issues/9

**Comment by migajek** [comment]
thank you, it was the misconfiguration indeed! Sorry for wasting your time with this issue ;)

BTW I believe it'd be great to let people know this extra configuration option is required. The main page of the project doesn't state that, and the nginx section link name (example configuration) made me think I don't need it, as I have fastcgi already configured :)

**Comment by xplicit** [comment]
The main page is about v0.4 and this issue is fixed in v0.4: hyperfastcgi reads keepalive setting from nginx and applies the correct one without additional option in config.

**Comment by migajek** [comment]
oh I see, indeed. That was me wrong again looking at the current branch README.md at github. Thanks!
